### DISCUSSION TITLE
Ideas - Text API for Summarization, Topics, and Sentiment

### DISCUSSION THREAD

**Discussion by incubatize** [deepgram/community#294] [Feature Requests]
Deepgram's DSLM-powered [Summarization](https://deepgram.com/learn/ai-speech-summarization-api) feature is now in GA release, supporting audio input. We'd love your feedback. 

We're also developing text endpoints for features like Summarization, Topic, Sentiment, and Intent.

We'd like to know your preferences:

- Would a single endpoint with query parameters (Example - `api.deepgram.com/v1/read?summarize=true&topics=true&sentiments=true` **Or** separate endpoints for each feature be more useful? Example - `api.deepgram.com/v1/read/summarize` `api.deepgram.com/v1/read/topics`

- What would be the typical length of your input text for our text endpoints?

- For Intent and Sentiment detection, do you want identification at the channel level or across all channels? If diarization is enabled, do you require speaker labels attached to identified intents or sentiments?

Please add your comments below. Let me know if you have any questions/need clarification. Thank you so much!

**Comment by micoach-agustin** [comment]
I would love to increase the characters of summarize in v2 for a longer audios
